src/main/java/org/spongepowered/api/command/Command.java

```diff
@@ -0,0 +1,312 @@
+
+        /**
+         * Determines what to do if a child command throws an exception.
+         *
+         * <p>Defaults to {@link ChildExceptionBehaviors#SUPPRESS}, which means
+         * that if a child command fails to execute, the error will be ignored
+         * and this base command will attempt to execute.</p>
+         *
+         * <p>See {@link ChildExceptionBehaviors} for other possible behaviors.
+         * </p>
+         *
+         * @param exceptionBehavior The {@link ChildExceptionBehavior} to adhere
+         *                          to.
+         * @return This builder, for chaining
+         */
+        Builder childExceptionBehavior(ChildExceptionBehavior exceptionBehavior);
+
+        /**
+         * Provides a simple description for this command, typically no more
+         * than one line.
+         *
+         * <p>Fuller descriptions should be provided through
+         * {@link #extendedDescription(Text)}</p>
+         *
+         * @param description The description to use, or {@code null} for no
+         *                    description
+         * @return This builder, for chaining
+         */
+        Builder description(@Nullable Text description);
+
+        /**
+         * Provides the logic of the command.
+         *
+         * <p>This is only optional if child commands are specified.</p>
+         *
+         * @param executor The {@link CommandExecutor} that will run the command
+         * @return This builder, for chaining
+         */
+        Builder executor(CommandExecutor executor);
+
+        /**
+         * Provides the description for this command.
+         *
+         * <p>A one line summary should be provided to
+         * {@link #description(Text)}</p>
+         *
+         * @param extendedDescription The description to use, or {@code null}
+         *                            for no description.
+         * @return This builder, for chaining
+         */
+        Builder extendedDescription(@Nullable Text extendedDescription);
+
+        /**
+         * The flags that this command should accept. See {@link Flags}.
+         *
+         * @param flags The {@link Flags} to accept
+         * @return This builder, for chaining
+         */
+        Builder flags(Flags flags);
+
+        /**
+         * Determines how an argument string should be split.
+         *
+         * <p>Defaults to splitting on spaces, ignoring spaces in quoted
+         * regions.</p>
+         *
+         * @param tokenizer The {@link InputTokenizer} to use
+         * @return This builder, for chaining
+         */
+        Builder inputTokenizer(InputTokenizer tokenizer);
+
+        /**
+         * The parameter set to use when parsing arguments. Parameters will be
+         * used in the order provided here.
+         *
+         * @param parameters The {@link Parameter}s to use
+         * @return This builder, for chaining
+         */
+        Builder parameters(Parameter... parameters);
```

Comment:
There are other areas in the command API that need to be updated for that too. Particularly for `Parameter.firstOf(...)` and `Parameter.seq(...)` - what I could do is make them look like

* `Parameter.firstParsingOf(param1).or(param2).or(param3).build()`
* `Parameter.parseSequenceOf(param1).then(param2).then(param3).build()`

and do this compromise.

(Names subject to change, the sequence one is somewhat rubbish)